src/components/Editor/EmptyLines.js

```diff
@@ -0,0 +1,75 @@
+// @flow
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import { Component } from "react";
+
+import actions from "../../actions";
+import { getSelectedSource } from "../../selectors";
+
+import getEmptyLines from "../../utils/parser/getEmptyLines";
```

Comment:
We should follow `setSymbols` pattern and add the empty lines to the AST store when a source is loaded:

https://github.com/devtools-html/debugger.html/blob/master/src/actions/sources.js#L382

This will help in two ways:

1. it's cached
2. we can update the breakpoint check to use a redux selector and not a css selector :)